bin/links.ml

```diff
@@ -51,6 +51,9 @@ let process_program ?(printer=print_value) (valenv, nenv, tyenv) (program, t) ex
   let tenv = (Var.varify_env (nenv, tyenv.Types.var_env)) in
 
   let (globals, _) = Backend.transform_program tenv program in
+
+  (if Settings.get_value BS.typecheck_only then exit 0);
```

Comment:
It would be cleaner to define a separate entry point for "type check ir only". Alas, `links.ml` is a big mess that could do with some Spring cleaning. Some future PR....

Reply:
I agree that links.ml needs some cleanup